src/writer.rs

```
@@ -571,14 +605,11 @@ impl<D: Distance> Writer<D> {
         let item_ids = self.item_indices(wtxn, options)?;
         self.delete_links_from_db(&item_ids, wtxn)?;
 
-        // 4. mark all nodes as updated
-        for item_id in item_ids {
-            self.database.remap_data_type::<Unit>().put(
-                wtxn,
-                &Key::updated(self.index, item_id),
-                &(),
-            )?;
-        }
+        // 4. Avoid marking all nodes as updated
+        assert!(
+            options.relink_all_items,
+            "forcing relinking of all items requires the relink_all_items option to be set to true"
+        );
```

maybe move this assert to the start of the method so we fail fast ?